plugins/callback/foreman.py

```diff
@@ -281,17 +333,37 @@ def send_reports(self, stats):
                 report['config_report']['status']['pending'] = total['changed']
                 report['config_report']['status']['applied'] = 0
 
-            self._send_data('report', host, report)
-
+            self._send_data('report', 'foreman', host, report)
             self.items[host] = []
 
+    def send_reports(self, stats):
+        if self.report_type == "foreman":
+            self.send_reports_foreman_legacy(stats)
+        elif self.report_type == "proxy":
+            self.send_reports_proxy_host_report(stats)
+        else:
+            self._display.warning(u'Unknown foreman endpoint type: {type}'.format(type=self.report_type))
+
+    def drop_nones(self, d: dict) -> dict:
```

Comment:
AFAIU this is not supported by python 2.7.5 or at least my (ansible 2.9.18) version. Do we want to make this compatible with older ansible versions?

Reply:
This *has* to run on Python 2.7 and Ansible 2.9.

Reply:
damn...

Reply:
Yep you are right, I rebased and it fails for 2.7. Damn, I am not good in Python. Let me try to figure this out.

Reply:
Ok thanks for help, rebased with 2.7 compatibility :-)